src/eos-updater-poll-common.c

```diff
@@ -0,0 +1,910 @@
+
+static void
+eos_update_info_dispose_impl (EosUpdateInfo *info)
+{
+  g_clear_pointer (&info->commit, g_variant_unref);
+  g_clear_object (&info->extensions);
+}
+
+static void
+eos_update_info_finalize_impl (EosUpdateInfo *info)
+{
+  g_free (info->checksum);
+  g_free (info->refspec);
+  g_free (info->original_refspec);
+  g_strfreev (info->urls);
+}
+
+EOS_DEFINE_REFCOUNTED (EOS_UPDATE_INFO,
+                       EosUpdateInfo,
+                       eos_update_info,
+                       eos_update_info_dispose_impl,
+                       eos_update_info_finalize_impl)
+
+EosUpdateInfo *
+eos_update_info_new (const gchar *checksum,
+                     GVariant *commit,
+                     const gchar *refspec,
+                     const gchar *original_refspec,
+                     gchar **urls,
+                     EosExtensions *extensions)
+{
+  EosUpdateInfo *info;
+
+  g_return_val_if_fail (checksum != NULL, NULL);
+  g_return_val_if_fail (commit != NULL, NULL);
+  g_return_val_if_fail (refspec != NULL, NULL);
+  g_return_val_if_fail (original_refspec != NULL, NULL);
+  g_return_val_if_fail (EOS_IS_EXTENSIONS (extensions), NULL);
+
+  info = g_object_new (EOS_TYPE_UPDATE_INFO, NULL);
+  info->checksum = g_strdup (checksum);
+  info->commit = g_variant_ref (commit);
+  info->refspec = g_strdup (refspec);
+  info->original_refspec = g_strdup (original_refspec);
+  info->urls = g_strdupv (urls);
+  info->extensions = g_object_ref (extensions);
+
+  return info;
+}
+
+static void
+eos_metadata_fetch_data_dispose_impl (EosMetadataFetchData *fetch_data)
+{
+  g_main_context_pop_thread_default (fetch_data->context);
+  g_clear_pointer (&fetch_data->context, g_main_context_unref);
+  g_clear_object (&fetch_data->task);
+}
+
+EOS_DEFINE_REFCOUNTED (EOS_METADATA_FETCH_DATA,
+                       EosMetadataFetchData,
+                       eos_metadata_fetch_data,
+                       eos_metadata_fetch_data_dispose_impl,
+                       NULL)
+
+EosMetadataFetchData *
+eos_metadata_fetch_data_new (GTask *task,
+                             EosUpdaterData *data,
+                             GMainContext *context)
+{
+  EosMetadataFetchData *fetch_data;
+
+  g_return_val_if_fail (G_IS_TASK (task), NULL);
+  g_return_val_if_fail (data != NULL, NULL);
+  g_return_val_if_fail (context != NULL, NULL);
+
+  fetch_data = g_object_new (EOS_TYPE_METADATA_FETCH_DATA, NULL);
+  fetch_data->task = g_object_ref (task);
+  fetch_data->data = data;
+  fetch_data->context = g_main_context_ref (context);
+
+  g_main_context_push_thread_default (context);
+  return fetch_data;
+}
+
+static gchar *
+cleanstr (gchar *s)
+{
+  gchar *read;
+  gchar *write;
+
+  if (s == NULL)
+    return s;
+
+  for (read = write = s; *read != '\0'; ++read)
+    {
+      /* only allow printable */
+      if (*read < 32 || *read > 126)
+        continue;
+      *write = *read;
+      ++write;
+    }
+  *write = '\0';
+
+  return s;
```

Comment:
It is taken from [eos-updater-util](https://github.com/endlessm/eos-updater/blob/master/src/eos-updater-util.c#L114) (again). Basically it is used on some strings gotten from either `/proc/device-tree/compatible` or from various files inside `/sys/class/dmi/id` directory. If there are non-printable characters we remove them. And we remove them, because we will use these values as a group name in the branch file (which is a GKeyFile).

At least, that is what I understood from the code. ;)